### PR TITLE
#554 Decoration issue fails due invalid project id when branch is in fork

### PR DESCRIPTION
**Pull request by kna0085**
Fix has been done using target_project_id instead of source_project_id.

### CODE REVIEW

File: src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabMergeRequestDecoratorIntegrationTest.java [hunk 1]
```diff
@@ -202,11 +203,11 @@ private void decorateQualityGateStatus(QualityGate.Status status) {
                 .willReturn(created()));
 
         wireMockRule.stubFor(post(urlPathEqualTo("/api/v4/projects/" + sourceProjectId + "/merge_requests/" + mergeRequestIid + "/discussions"))
-                .withRequestBody(equalTo("body=summary+comm%C3%A9nt%0A%0A%5Blink+text%5D"))
+                .withRequestBody(equalTo("body=summary+comm%C3%83%C2%A9nt%0A%0A%5Blink+text%5D"))
```
Comment (mc1arke) [inline src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabMergeRequestDecoratorIntegrationTest.java:206]: I can't see a reason why this line would need to change. This implies that the comment is being double encoded which shouldn't happen as part of your changes.

File: src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabMergeRequestDecoratorIntegrationTest.java [hunk 2]
```diff
@@ -202,11 +203,11 @@ private void decorateQualityGateStatus(QualityGate.Status status) {
...
                 .willReturn(created().withBody(discussionPostResponseBody(discussionId, discussionNote(noteId, user, "summary comment", true, false)))));
 
         wireMockRule.stubFor(post(urlPathEqualTo("/api/v4/projects/" + sourceProjectId + "/merge_requests/" + mergeRequestIid + "/discussions"))
-                .withRequestBody(equalTo("body=issu%C3%A9&" +
+                .withRequestBody(equalTo("body=issu%C3%83%C2%A9&" +
```
Comment (mc1arke) [inline src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabMergeRequestDecoratorIntegrationTest.java:210]: This also implies double-encoding which wouldn't be right

**Review by mc1arke — CHANGES_REQUESTED** [review]
I'm happy with the concept, although you have test failures, and the changes within your tests don't quite seem right.

Could you also rebase your changes onto the latest `master` branch please?

File: src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabMergeRequestDecoratorIntegrationTest.java (hunk 1 above)
Comment (johnou) [reply src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabMergeRequestDecoratorIntegrationTest.java:206]: should be fixed with https://github.com/mc1arke/sonarqube-community-branch-plugin/pull/580

Comment (kna0085) [reply src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabMergeRequestDecoratorIntegrationTest.java]: @mc1arke test issue was solved.

File: src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabMergeRequestDecoratorIntegrationTest.java (hunk 2 above)
Comment (kna0085) [reply src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabMergeRequestDecoratorIntegrationTest.java]: this one was solved too

**Comment by kna0085** [comment]
@mc1arke rebase and test fixes were uploaded. However, I'm seeing that analisys is saying there are conflicts on the file I've changed but I see the file and it seems to be ok. could you point me out to what is wrong with it?

**Comment by mc1arke** [comment]
Merged. Thanks for the contribution!

**Pull request merged**